osu.Game/Graphics/Containers/ColouredLinkFlowContainer.cs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Sprites;
+using osu.Game.Online.Chat;
+using osuTK.Graphics;
+
+namespace osu.Game.Graphics.Containers
+{
+    public class ColouredLinkFlowContainer : LinkFlowContainer
+    {
+        private Color4 idleColour;
+
+        public Color4 IdleColour
+        {
+            get => idleColour;
+            set
+            {
+                idleColour = value;
+                foreach (var linkCompiler in InternalChildren.OfType<ColouredDrawableLinkCompiler>())
+                    linkCompiler.IdleColour = value;
+            }
+        }
+
+        private Color4 hoverColour;
+
+        public Color4 HoverColour
+        {
+            get => hoverColour;
+            set
+            {
+                hoverColour = value;
+                foreach (var linkCompiler in InternalChildren.OfType<ColouredDrawableLinkCompiler>())
+                    linkCompiler.HoverColour = value;
+            }
+        }
```

Comment:
Yeah, definitely not a fan of these.

> I also tried to experiment with using `Bindable`s here, which might've been a cleaner solution to the first problem described above, but couldn't get them to work as intended.

What was the problem?

Reply:
Couldn't get colour value propagation to work properly, though that was probably just my lack of experience with bindables, and I didn't want to spend too much time on something that I wasn't sure will work. I'll try to approach that again later.

Reply:
After a short look at the situation the protected-ness of `{Idle,Hover}Colour` is the main problem here. If the base classes do defaults in BDL then there's really no good way to reliably override them. Maybe what is desired here is something that makes the default more explicit, i.e.

```csharp
class OsuHoverContainer
{
        [Resolved]
        private OsuColour colours { get; set; };

        private Color4? hoverColour;

        protected Color4 HoverColour
        {
            get => hoverColour ?? colours.Yellow;
            set => hoverColour = value;
        }

}
```

and the same thing for `IdleColour`. Might be better off as a separate PR, though?